src/test/java/net/sf/jabref/importer/CustomImporterTest.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    @Test
+    public void testGetBasePathUrl() throws Exception {
+        assertEquals(new File("src/main/java/net/sf/jabref/importer/fileformat/CopacImporter.java").toURI().toURL(), importer.getBasePathUrl());
+    }
+
+    @Test
+    public void testGetAsStringList() {
+        assertEquals("Copac", importer.getAsStringList().get(0));
+        assertEquals("cpc", importer.getAsStringList().get(1));
+        assertEquals("net.sf.jabref.importer.fileformat.CopacImporter", importer.getAsStringList().get(2));
+        assertEquals("src/main/java/net/sf/jabref/importer/fileformat/CopacImporter.java", importer.getAsStringList().get(3));
+    }
+
+    @Test
+    public void testEqualsTrue() {
+        assertEquals(importer, importer);
+    }
+
+    @Test
+    public void testEqualsFalse() {
+        assertNotEquals(new CopacImporter(), importer);
+    }
+
+    @Test
+    public void testCompareToSmaller() {
+        CustomImporter ovidImporter = new CustomImporter(new OvidImporter());
+
+        assertTrue(importer.compareTo(ovidImporter) < 0);
+    }
+
+    @Test
+    public void testCompareToGreater() {
+        CustomImporter bibtexmlImporter = new CustomImporter(new BibTeXMLImporter());
+        CustomImporter ovidImporter = new CustomImporter(new OvidImporter());
+
+        assertTrue(ovidImporter.compareTo(bibtexmlImporter) > 0);
+    }
+
+    @Test
+    public void testCompareToEven() {
+        assertEquals(0, importer.compareTo(importer));
+    }
+
+    @Test
+    public void testToString() {
+        assertEquals("Copac", importer.toString());
+    }
+
+    @Test
+    public void testClassicConstructor() {
+        CustomImporter customImporter = new CustomImporter("Copac", "cpc", "net.sf.jabref.importer.fileformat.CopacImporter", "src/main/java/net/sf/jabref/importer/fileformat/CopacImporter.java");
+
+        assertEquals(customImporter.getName(), importer.getName());
+        assertEquals(customImporter.getClidId(), importer.getClidId());
+        assertEquals(customImporter.getBasePath(), importer.getBasePath());
+        assertEquals(customImporter.getClassName(), importer.getClassName());
+    }
+
+    @Test
+    public void testListConstructor() {
+        List<String> dataTest = Arrays.asList("Ovid", "ovid", "net.sf.jabref.importer.fileformat.OvidImporter", "src/main/java/net/sf/jabref/importer/fileformat/OvidImporter.java");
+        CustomImporter customImporter = new CustomImporter(dataTest);
+        CustomImporter customOvidImporter = new CustomImporter(new OvidImporter());
+
+        assertEquals(customImporter.getName(), customOvidImporter.getName());
+        assertEquals(customImporter.getClidId(), customOvidImporter.getClidId());
+        assertEquals(customImporter.getBasePath(), customOvidImporter.getBasePath());
+        assertEquals(customImporter.getClassName(), customOvidImporter.getClassName());
```

Comment:
Can't this replaced by equals? Also I think `equals` needs a rework, it only checks for the type of `CustomImporter` and its name.